tests/linearGaussian_npse_test.py

```diff
@@ -179,27 +182,94 @@ def test_npse_iid_inference(num_trials):
     prior_mean = zeros(num_dim)
     prior_cov = eye(num_dim)
     prior = MultivariateNormal(loc=prior_mean, covariance_matrix=prior_cov)
-    gt_posterior = true_posterior_linear_gaussian_mvn_prior(
-        x_o, likelihood_shift, likelihood_cov, prior_mean, prior_cov
-    )
-    target_samples = gt_posterior.sample((num_samples,))
 
-    inference = NPSE(prior, show_progress_bars=True)
+    # This check that our method to handle "general" priors works.
+    # i.e. if NPSE does not get a proper passed by the user.
+    prior_npse = prior if not with_improper_prior else None
+    inference = NPSE(prior_npse, show_progress_bars=True, sde_type=sde_type)
 
     theta = prior.sample((num_simulations,))
     x = linear_gaussian(theta, likelihood_shift, likelihood_cov)
 
     score_estimator = inference.append_simulations(theta, x).train(
-        training_batch_size=100,
+        training_batch_size=200, max_num_epochs=400
     )
-    posterior = inference.build_posterior(score_estimator)
-    posterior.set_default_x(x_o)
-    samples = posterior.sample((num_samples,))
+    for iid_method in iid_methods:
+        for num_trial in num_good_evals_per_method[iid_method]:
+            x_o = zeros(num_trial, num_dim)
+            posterior = inference.build_posterior(score_estimator)
+            posterior.set_default_x(x_o)
+            samples = posterior.sample((num_samples,), iid_method=iid_method)
+
+            gt_posterior = true_posterior_linear_gaussian_mvn_prior(
+                x_o, likelihood_shift, likelihood_cov, prior_mean, prior_cov
+            )
+            target_samples = gt_posterior.sample((num_samples,))
+
+            # Compute the c2st and assert it is near chance level of 0.5.
+            # Some degradation is expected, also because posterior get tighter which
+            # usually makes the c2st worse.
+            check_c2st(
+                samples,
+                target_samples,
+                alg=f"npse-vp-gaussian-2D-{iid_method}-{num_trial}iid-trials",
+                tol=0.25 + 0.05 * with_improper_prior,
+            )
 
-    # Compute the c2st and assert it is near chance level of 0.5.
-    check_c2st(
-        samples, target_samples, alg=f"npse-vp-gaussian-1D-{num_trials}iid-trials"
+
+@pytest.mark.slow
```

Comment:
this is essentially the same test as above but with `prior=Uniform` no? So I suggest merging the two tests and just using `pytest.mark.parametrize("prior", ("gaussian", "uniform", None))`

or am I missing something?